q2_quality_control/tests/test_decontam.py

```diff
@@ -121,26 +126,58 @@ def setUp(self):
             self.get_data_path('expected/decon_default_score_table.qza'))
         self.identify_table = id_table.view(qiime2.Metadata)
 
+        seq_removal_table = qiime2.Artifact.load(
+            self.get_data_path('expected/remove_contam_test_table.qza'))
+        self.seq_removal_asv_table = seq_removal_table.view(
+            qiime2.Metadata).to_dataframe()
+        seq_removal_rep_seqs = qiime2.Artifact.load(
+            self.get_data_path('expected/remove_contam_rep_seqs.qza'))
+        self.rep_seq_removal_table = \
+            seq_removal_rep_seqs.view(qiime2.Metadata)
+        seq_removal_rep_seq_decontam_scores = qiime2.Artifact.load(
+            self.get_data_path(
+                'expected/remove_contam_test_identify_scores.qza'))
+        self.rep_seq_decon_scores =  \
+            seq_removal_rep_seq_decontam_scores.view(qiime2.Metadata)
+
     def test_remove(self):
-        exp_table = pd.read_csv(
-            self.get_data_path('expected/no-contaminant-asv-table.tsv'),
-            sep='\t', index_col=0)
-        output_asv_table = decontam_remove(
-            table=self.asv_table,
-            decontam_scores=self.identify_table,
-            threshold=0.1)
-        temp_table = output_asv_table.to_dataframe()
+        rep_seq_output = qiime2.Artifact.load(
+            self.get_data_path('expected/output_remove_contam_rep_seqs.qza'))
+        self.rep_seq_output = rep_seq_output.view(
+            qiime2.Metadata).to_dataframe()
+        output_table = qiime2.Artifact.load(
+            self.get_data_path('expected/output_remove_contam_table.qza'))
+        self.rep_seq_output_table = output_table.view(
+            qiime2.Metadata).to_dataframe()
+        output_asv_table, output_rep_seqs = decontam_remove(
+            table=self.seq_removal_asv_table,
+            decontam_scores=self.rep_seq_decon_scores,
+            threshold=0.1,
+            rep_seqs=self.rep_seq_removal_table)
+        temp_table = output_asv_table.to_dataframe().transpose()
+        temp_table.index.name = 'id'
+        temp_table.columns = range(1, len(temp_table.columns) + 1)
```

Comment:
I think the reason that your ids may not be lining up is because the input and output feature tables share no features. I'm submitting some edits that I made where I broke your .qza files out for easier access in tests, and when I do the following:

```bash
(q2dev-amplicon) ➜  expected git:(gregs-review-pr81) ✗ biom table-ids --observations -i remove_contam_test_table/feature-table.biom | sort | head
01173073677a287321be3484fbed0007
0160e14a78b18b903618f11bc732746e
01b99cb344ed2530f7d80897ffe257a9
01ce91fd8dbecf637eb5e67cdab5c5aa
01e0b7ac306895be84179f2715af269b
025dd300b0ccbd9d5898969c2a1ab138
02878fe3ccc81d4c884ca5574178d6a0
029cc71dc93341de90188b686798aa0d
02ef9a59d6da8b642271166d3ffd1b52
0305a4993ecf2d8ef4149fdfc7592603

(q2dev-amplicon) ➜  expected git:(gregs-review-pr81) ✗ biom table-ids --observations -i output_remove_contam_table/feature-table.biom | sort | head
003d45b24e8a58c0b20bf2bbfd7ff87f
008b8e3f9166408dc781796afd64a959
00dff50b45156cbb7debd5b5e82bf565
010b4e6583bfda219e2967b9b3973de0
010d323d97cc7ed0cd0081ec591d051a
012a37c5b111ebc368d03345b6049198
01d402b3bf484b51373700f1c1537b4b
025ebdf9e8907de3cb0010efa7424798
026a7407ef9739f4505fde62e29e6795
031b2591f8ff1aca82a3698fd490988c
```

you can see that the feature ids don't seem to align. 

So I think you thought this was an issue with your test, but it was actually an issue with your test data.